tron/utils/persistable.py

```diff
@@ -13,5 +13,10 @@ def to_json(state_data: Dict[Any, Any]) -> Optional[str]:
 
     @staticmethod
     @abstractmethod
-    def from_json(state_data: str) -> Dict[Any, Any]:
+    def from_json(state_data: str) -> Dict[str, Any]:
+        # This method is called on because it is intended to handle the deserialization of JSON data into a
+        # dictionary representation of the state. This allows the method to be used in a more flexible and generic way,
+        # enabling different classes to implement their own specific logic for converting the dictionary into an instance of the
+        # class. By returning a dictionary, the method provides a common interface for deserialization, while allowing subclasses
+        # to define how the dictionary should be used to restore the state of the object.
```

Comment:
iiuc from a previous chat about the serde we're doing with this project, we're currently returning a dict to make it easier to fit in alongside the pickle deserialization logic, right?

(if not and there's no blockers to doing this, `from_json()` should return the actual instance since then we skip dealing with formatting the intermediate dict that we get from the json string)

Reply:
I think so? to_json takes a dictionary and returns a string so when I was implementing from_json I was trying to get it to return the same format it was in before

Reply:
yea, to_json will eventually change once the pickle code is gone: see the thread starting at https://github.com/Yelp/Tron/pull/997#discussion_r1769247550

if possible with the current code flow, having from_json() be a classmethod that returns the deserialized object would be cleaner/closer to the final desired state of all this serde code :)

Reply:
(i.e., by "if possible", i mean if we don't need to make any large changes from how the deserialization works with the pickle fallback - e.g., if the pickle restore doesn't return the deserialized object and instead returns the binary data that is then loaded, this is probably fine until we also update to_json :))